generated/nidaqmx/_time.py

```diff
@@ -0,0 +1,46 @@
+from __future__ import annotations
+
+import pytz
+from tzlocal import get_localzone
+from datetime import timezone
+from datetime import datetime as std_datetime
+from hightime import datetime as ht_datetime
+from typing import Optional, Union
+
+# theoretically the same as astimezone(), but with support for dates before 1970
+def _convert_to_desired_timezone(expected_time_utc: Union[std_datetime, ht_datetime], tzinfo: Optional[timezone] = None):
+    # if timezone matches, no need to do conversion
+    if expected_time_utc.tzinfo is tzinfo:
+        return expected_time_utc
+
+    # if timezone is not defined, use system timezone
+    if tzinfo is None:
+        local_timezone = get_localzone()
+        tzinfo = pytz.timezone(str(local_timezone))
```

Comment:
This uses `tzlocal.get_localzone()`, which returns a `zoneinfo.ZoneInfo`. Then it uses `pytz.timezone()` to convert the `ZoneInfo` into a `datetime.tzinfo`. However, [zoneinfo](https://docs.python.org/3/library/zoneinfo.html) says

> [ZoneInfo](https://docs.python.org/3/library/zoneinfo.html#zoneinfo.ZoneInfo) is a concrete implementation of the [datetime.tzinfo](https://docs.python.org/3/library/datetime.html#datetime.tzinfo) abstract base class, and is intended to be attached to tzinfo, either via the constructor, the [datetime.replace](https://docs.python.org/3/library/datetime.html#datetime.datetime.replace) method or [datetime.astimezone](https://docs.python.org/3/library/datetime.html#datetime.datetime.astimezone):

Do you really need `pytz`? Based on the docs, I think you can set `tzinfo = get_localzone()`.